app/src/main/java/com/onegini/mobile/exampleapp/view/activity/DashboardActivity.java

```diff
@@ -73,20 +75,25 @@ public void onSuccess() {
 
           @Override
           public void onError(final OneginiLogoutError oneginiLogoutError) {
-            handleLogoutError(oneginiLogoutError);
+            handleLogoutError(oneginiLogoutError, userProfile);
           }
         }
     );
   }
 
-  private void handleLogoutError(final OneginiLogoutError oneginiLogoutError) {
+  private void handleLogoutError(final OneginiLogoutError oneginiLogoutError, final UserProfile userProfile) {
     @OneginiLogoutError.LogoutErrorType final int errorType = oneginiLogoutError.getErrorType();
     if (errorType == OneginiLogoutError.LOCAL_LOGOUT) {
       showToast("The user was only logged out on the device. The access token has not been invalidated on the server-side.");
     } else if (errorType == OneginiLogoutError.GENERAL_ERROR) {
       // General error handling for other, less relevant errors
       showToast("Logout error: " + oneginiLogoutError.getErrorDescription());
+    } else if (errorType == OneginiLogoutError.DEVICE_DEREGISTERED) {
+      new DeregistrationUtil(this).onDeviceDeregistered();
+    } else if (errorType == OneginiLogoutError.USER_DEREGISTERED) {
+      new DeregistrationUtil(this).onUserDeregistered(userProfile);
```

Comment:
I think that you marked wrong place. We don't call onUserDeregistered on successful logout. But indeed we can do what you propose for deregistration flow.